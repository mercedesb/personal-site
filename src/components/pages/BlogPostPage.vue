<template>
   <div>
     <header :class="'ContentHeader ContentHeader--' + color">
      <div class="ContentHeader-text">
        <h1>{{ page.title }}</h1>
        <h5>{{ formattedPublishDate }}</h5>
      </div>
    </header>
    <main v-if="page.mainContent" class='FlexContainer FlexContainer--justifyCenter'>
      <p class="PageContent">
        <parse-markdown :source="page.mainContent" />
      </p>
    </main>
  </div>
</template>

<script>
const moment = require('moment')

import PageHeader from '../PageHeader.vue'
import CTALink from '../CTALink.vue'

export default {
  components: {
    PageHeader, CTALink
  },
  props: {
    color: {
      type: String,
      default: 'gray'
    }
  },
  computed: {
    page() {
      return this.$store.state.blogPost
    },
    formattedPublishDate() {
      return moment(this.date).format('MMM DD YYYY')
    }
  },
  created () {
    this.$store.dispatch('getBlogPost', this.$route.params.urlSegment)
  }
}
</script>

<style lang="scss">
$header-height: 200px;

.ContentHeader {
  box-shadow: $base-drop-shadow;
  max-height: $header-height;
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;

  &-text {
    margin: $small-spacing $base-spacing;
    overflow: hidden;
    text-align:center;
  }

  @include background-color;
}
</style>