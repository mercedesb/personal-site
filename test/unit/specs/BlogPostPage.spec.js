import BlogPostPage from '@/components/pages/BlogPostPage.vue'
import { TestUtility } from './TestUtility'

const mockFormat = jest.fn(() => 'Apr 18 2018')

jest.mock('moment', () => () => ({
  format: mockFormat
}))

describe('BlogPostPage', () => {
  let component

  const initialProps = {
    color: 'blue'
  }

  const store = {
    state: {
      blogPost: TestUtility.blogPosts[0],
      navLinks: []
    }
  }

  const shallow = propsData => TestUtility.shallow(BlogPostPage, {
    store,
    propsData: {
      ...initialProps,
      ...propsData
    }
  })

  describe('Snapshots', () => {
    describe('with a color passed in', () => {
      it('matches snapshot', () => {
        component = shallow()
        expect(component.element).toMatchSnapshot()
      })
    })

    describe('without a color passed in', () => {
      it('matches snapshot', () => {
        component = shallow({color: undefined})
        expect(component.element).toMatchSnapshot()
      })
    })
  })

  describe('Computed', () => {
    describe('page', () => {
      beforeEach(() => {
        component = shallow()
      })

      it('has page populated', () => {
        expect(component.vm.page).toBeDefined()
      })
    })

    describe('formattedPublishDate', () => {
      it('calls moment().format', () => {
        component = shallow()
        expect(mockFormat).toHaveBeenCalled()
      })
    })
  })

  describe('Lifecycle', () => {
    describe('created', () => {
      it('dispatches getBlogPost to the store', () => {
        component = TestUtility.mount(BlogPostPage, {
          store,
          propsData: {
            ...initialProps
          }
        })
        const expectedPath = TestUtility.mockRoute
        expect(component.vm.$store.dispatch).toHaveBeenCalledWith('getBlogPost', expectedPath)
      })
    })
  })
})
