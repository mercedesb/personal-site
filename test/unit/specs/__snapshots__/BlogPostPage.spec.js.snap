// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostPage Snapshots with a color passed in matches snapshot 1`] = `
<div>
  <header
    class="ContentHeader ContentHeader--blue"
  >
    <div
      class="ContentHeader-text"
    >
      <h1>
        blog post 1 title
      </h1>
       
      <h5>
        Apr 18 2018
      </h5>
    </div>
  </header>
   
  <!---->
</div>
`;

exports[`BlogPostPage Snapshots without a color passed in matches snapshot 1`] = `
<div>
  <header
    class="ContentHeader ContentHeader--gray"
  >
    <div
      class="ContentHeader-text"
    >
      <h1>
        blog post 1 title
      </h1>
       
      <h5>
        Apr 18 2018
      </h5>
    </div>
  </header>
   
  <!---->
</div>
`;
